### DISCUSSION TITLE
Is PixiJS the right tool for our project? A web-based SVG drawing tool

### DISCUSSION THREAD

**Discussion by CallMeTuesday** [pixijs/pixijs#9260] [Q&A]
We've been building a web app using PaperJS specifically for lasso painting and have hit a wall. After about a thousand path objects, which sounds like a lot but is not when you're painting, the app slows down tremendously. We're looking at ways to improve the performance and have considered using Two.JS, Snap, Pixi, etc. While the desired output is SVG, users will be able to export as rasters as well. However, we want to maintain the ability to select, delete, transform the path elements in addition to giving the user the control to rasterize vs doing it for them. Additionally, we want to maintain visual clarity as much as possible, and raster gets funky when you scale it.

How does Pixi handle thousands of SVG path elements? 

I have a hosted example I can share but it's not public yet. I'd be happy do DM it to whomever is interested in helping with this

Here is an example SVG file created with our app

![lassoh-painting 2023-03-10 17-31-11](https://user-images.githubusercontent.com/713101/224503126-a81dd9ea-7a56-423a-8f83-3b16b8957524.svg)

As you can see, there aren't that many elements. I believe the total path objects = ~1400 and around 13K points

**Comment by bigtimebuddy** [comment]
PixiJS has built-in support for rasterizing SVG elements and there are a few 3rd party plugins (e.g., [pixi-svg](https://npmjs.com/packages/pixi-svg), [@pixi-essentials/svg](https://www.npmjs.com/package/@pixi-essentials/svg), etc) which can convert SVG elements into PixiJS's vector graphics (i.e., [Graphics](https://pixijs.io/guides/basics/graphics.html) objects).

I would say that any WebGL-based rendering is going to give you better performance results, PixiJS or another library. Your challenge will be how to manage your texture memory effectively, but 13K points is not terrible and will perform much better than the DOM.
